R/redundant_ifelse_linter.R

```diff
@@ -46,17 +46,17 @@ redundant_ifelse_linter <- function(allow10 = FALSE) {
     if (!allow10) {
       num_expr <- xml2::xml_find_all(xml, num_xpath)
       matched_call <- xp_call_name(num_expr)
-      # [1] call; [2] logical condiditon
+      # [1] call; [2] logical condition
       first_arg <- xml2::xml_find_chr(num_expr, "string(expr[3]/NUM_CONST)")
       second_arg <- xml2::xml_find_chr(num_expr, "string(expr[4]/NUM_CONST)")
-      replacement <- ifelse(
-        first_arg %in% c("0", "1") | second_arg %in% c("0", "1"),
-        "as.numeric",
-        "as.integer"
-      )
+      is_numeric_01 <- first_arg %in% c("0", "1") | second_arg %in% c("0", "1")
```

Comment:
I would think a flow like

ifelse(is_numeric_01, "as.numeric", "as.integer")
is_negated <- first_arg %in% c("0", "0L")
ifelse(is_negated, "!x", "x")

then build the message with those two bits

would be simpler/easier to follow, is there a reason that won't work?

Reply:
Thanks for the suggestion! 

That totally works; I was just being stingy in creating objects. But I guess this is a bit more readable.

Reply:
if we really want to skimp we don't need to allocate either is_* variable since they're not used only once. either ways basically don't though. these objects are almost always tiny (maybe a couple dozen elements in the most extreme realistic case)